kore/src/Kore/Logger/DebugAxiomEvaluation.hs

```diff
@@ -0,0 +1,231 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+
+-}
+
+module Kore.Logger.DebugAxiomEvaluation
+    ( AxiomEvaluationState (..)
+    , DebugAxiomEvaluation (..)
+    , DebugAxiomEvaluationOptions (..)
+    , filterDebugAxiomEvaluation
+    , parseDebugAxiomEvaluationOptions
+
+    -- * logging functions. Import qualified.
+    , end
+    , notEvaluated
+    , reevaluation
+    , start
+    ) where
+
+import Control.Applicative
+    ( Alternative (..)
+    )
+import Data.Function
+    ( on
+    )
+import Data.Maybe
+    ( catMaybes
+    , fromMaybe
+    )
+import Data.Set
+    ( Set
+    )
+import qualified Data.Set as Set
+import Data.Text
+    ( Text
+    )
+import qualified Data.Text as Text
+    ( pack
+    )
+import Data.Text.Prettyprint.Doc
+    ( Pretty
+    )
+import qualified Data.Text.Prettyprint.Doc as Pretty
+import Data.Typeable
+import Options.Applicative
+    ( Parser
+    )
+import qualified Options.Applicative as Options
+
+import Kore.Logger
+    ( Entry (fromEntry, toEntry)
+    , LogAction (LogAction)
+    , MonadLog
+    , Severity (..)
+    , SomeEntry
+    , logM
+    , unLogAction
+    )
+import qualified Kore.Logger as Log.DoNotUse
+import Kore.Step.Axiom.Identifier
+    ( AxiomIdentifier
+    )
+
+{- | A log 'Entry' when a rule is applied.
+
+We will log the applied rule and its unification or matching condition.
+
+ -}
+data DebugAxiomEvaluation =
+    DebugAxiomEvaluation
+    { identifier :: !(Maybe AxiomIdentifier)
+    , secondaryIdentifier :: !(Maybe Text)
+    , state :: !AxiomEvaluationState
+    , severity :: !Severity
```

Comment:
Wouldn't the `Severity` always be `Debug`?

Reply:
Sometimes, i.e. for the symbols selected by `--debug-simplification-axiom`, it's "Info". That's how I interpreted our chat about this:

"We want to log all these things at the Debug level, but have options to print the entries for specific functions regardless of the --log-level selected by the user."

I guess that I should have used "Critical" to really print entries regardless of the `--log-level`, but I thought I'd try "Info" first. Should I log them as "Critical" instead?

Reply:
That's OK for now.